qiskit/extensions/qiskit_simulator/noise.py

```diff
@@ -65,6 +66,15 @@ def noise(self, m, q):
 QuantumCircuit.noise = noise
 CompositeGate.noise = noise
 
-# Add to QASM header for parsing
-QuantumCircuit.header += "\ngate noise(m) a {}" + \
-    "  // (local_qiskit_simulator) switch noise off (0) or on (1)"
+
+# switch noise off (0) or on (1) (identity)
+QuantumCircuit.definitions["noise"] = {
+    "print": True,
+    "opaque": False,
+    "n_args": 1,
+    "n_bits": 1,
+    "args": ["m"],
```

Comment:
Could we have a more descriptive name than "m"? :) (or a comment at least with a description at least)